src/main/java/com/ticticboooom/twerkitmeal/TwerkItMeal.java

```diff
@@ -133,6 +154,46 @@ private List<BlockPos> getNearestBlocks(World world, BlockPos pos) {
             return list;
         }
 
+        private List<AgeableEntity> getNearestAgeableEntities(World world, BlockPos pos) {
+            List<AgeableEntity> list = new ArrayList<>();
+
+            for (int x = -TwerkConfig.effectRadius; x <= TwerkConfig.effectRadius; x++)
+                for (int y = -2; y <= 2; y++)
+                    for (int z = -TwerkConfig.effectRadius; z <= TwerkConfig.effectRadius; z++) {
+                        List<AgeableEntity> posEntitiesList =
+                                world.getEntitiesWithinAABB(
```

Comment:
Just use an AxisAlignedBB for the whole area instead of block by block. This can replace your whole method.

Reply:
Haven't tested yet, but could work as well.
I thought of block by block because inside a same block ther can be multiple entities.
All in all, this is my first attempt at modding Minecraft. I am still adapting to the API. Any suggestions are welcome of course